honeybadgermpc/mimc.py

```diff
@@ -0,0 +1,64 @@
+from math import log, ceil
+import logging
+import asyncio
+from honeybadgermpc.mpc import PreProcessedElements, TaskProgramRunner, Subgroup
+from honeybadgermpc.field import GF
+from honeybadgermpc.mixins import BeaverTriple, MixinOpName
+
+# ROUND: iteration time of MiMC encryption function, r = ceil(log(p, 3)) = 161
+ROUND = ceil(log(Subgroup.BLS12_381))
+
+
+async def mimc(context, x, k):
+    # message x \in Fp, secret key k \in Fp
+
+    pp_elements = PreProcessedElements()
+
+    # def cube(): [x] -> [x^3]
+    async def cube(x):
+        # first compute a tuple ([r], [r^2], [r^3])
+        r = pp_elements.get_rand(context)
+        r2 = await (r * r)
+        r3 = await (r * r2)
+
+        y = await (x-r).open()
+        x3 = 3*y*r2 + 3*(y**2)*r + y**3 + r3
```

Comment:
I think documenting this with a comment like `[x^3] = 3y[r^2] + 3y^2[r] + y^3 + [r^3]` would be a good idea for future maintenance.

Reply:
Resolved.